pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PanDomain.scala

```diff
@@ -8,7 +8,7 @@ object PanDomain {
   /**
    * Check the authentication status of the provided credentials by examining the signed cookie data.
    */
-  def authStatus(cookieData: String, publicKey: PublicKey, validateUser: AuthenticatedUser => Boolean = guardianValidation): AuthenticationStatus = {
+  def authStatus(cookieData: String, publicKey: PublicKey, validateUser: AuthenticatedUser => Boolean): AuthenticationStatus = {
```

Comment:
I think this is a pretty important change for us. If tools were relying on this default argument then their security will break when they upgrade.

It might be safer to keep this as is and introduce a new function that doesn't include our check?

Reply:
Hmm I'm torn. You're right, the safest option is not to change behaviour but the existing code is confusing because there are essentially two `validateUser` functions: one you can customise and this one that you can't. 

I found this code in [mobile-entry](https://github.com/guardian/mobile-entry/blob/master/app/ext/EntryAuthActions.scala#L112) that works-around this in a pretty ugly way.

Adding a third function would increase the tech debt in the library, although I guess we could minimise it by adding deprecated annotations.

Can we catch up quickly IRL about this? I've done an audit of our `validateUser` functions on GitHub so that can feed into the discussion.

Reply:
Yep, let's converse in the real world.